File: backend/QuizHubApplication/DTOs/Responses/QuizDetailsDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuizHubApplication.DTOs.Responses
{
    public record QuizDetailsDto(int Id, string Title, string? Description, int TimeLimit, string DifficultyLevel,string CategoryName, string? CategoryDescription,
        List<QuestionDto>Questions)
    {
    }
}

File: frontend/QuizHubFrontend/src/app/features/quiz/models/QuizDetailsDto.ts
import { QuestionDto } from './QuestionDto';

export interface QuizDetailsDto {
    id: number;
    title: string;
    description?: string; // ? jer može biti null
    timeLimit: number;
    difficultyLevel: string;
    categoryName: string;
    categoryDescription?: string; // ? jer može biti null
    questions: QuestionDto[];
}
